docs/wiki/CL/overview.md

```diff
@@ -1,20 +1,415 @@
+
+The central challenge of the Byzantine Generals’ Problem is that the messages can get somehow delayed, destroyed or lost. In addition, even if a message is successfully delivered, one or more generals may choose (for whatever reason) to act maliciously and send a fraudulent message to confuse the other generals, leading to a total failure. If we apply the dilemma to the context of blockchains, each general represents a network node, and the nodes need to reach consensus on the current state of the system. Putting in another way, the majority of participants within a distributed network have to agree and execute the same action in order to avoid complete failure. 
+
+Therefore, the only way to achieve consensus in these types of distributed system is by having at least ⅔ or more reliable and honest network nodes. This means that if the majority of the network decides to act maliciously, the system is susceptible to failures and attacks (such as the 51% attack).
+
+In blockchain and distributed ledger systems, the Byzantine Generals' Problem is analogous to nodes (validators or miners) needing to agree on the state of the ledger despite some nodes potentially acting maliciously.
+
+
+## Introduction to Consensus Layer (CL)
+
+**tldr;**
+
+> - Consensus is a way to build reliable distributed systems with unreliable components.
+> - Blockchain-based distributed systems aim to agree on a single history of transactions.
+> - Proof-of-Work and Proof-of-stake are not consensus protocols, but enable consensus protocols.
+> - Nodes and validators are the actors of the consensus system.
+> - Slots and epochs regulate consensus time.
+> - Blocks and attestations are the currency of consensus.
+
+The Consensus Layer (CL) is a fundamental component that ensures the network's security, reliability, and efficiency. Originally, Ethereum utilized Proof-of-Work (PoW) as its consensus mechanism, similar to Bitcoin. PoW, while effective in maintaining decentralization and security, has significant drawbacks, including high energy consumption and limited scalability. To address these issues, Ethereum has transitioned to Proof-of-Stake (PoS), a more sustainable and scalable consensus mechanism.
+
+The Ethereum network consists of many individual nodes. Each node operates independently and communicates over the Internet, which is often unreliable and asynchronous. A node might always behave correctly, or it might be faulty in various ways: being offline, using a different protocol version, trying to mislead other nodes, sending contradictory messages, or experiencing other issues.
```

Comment:
I feel like this was also explained elsewhere, maybe it could be shorter or reference the protocol overview.

Reply:
I removed it from the initial the BFT explanation